pyproject.toml

```diff
@@ -0,0 +1,2 @@
+[tool.pyright]
+extraPaths = [ "discovery-infra" ]
```

Comment:
What's the meaning of this file?

Reply:
I explain this in the commit message

Reply:
For information about `pyrpoject.toml` in general see https://www.python.org/dev/peps/pep-0621/

Here I'm just adding metadata for the `pyright` tool, so if anybody wants to use pyright on this project, it will work properly. I'm using `pyright` as my Python LSP for Neovim